java/src/main/java/com/logicalclocks/hsfs/constructor/Query.java

```diff
@@ -303,4 +303,17 @@ public Query appendFeature(Feature feature) {
     this.leftFeatures.add(feature);
     return this;
   }
+
+  public boolean isTimeTravel() {
+    if (leftFeatureGroupStartTime != null || leftFeatureGroupEndTime != null) {
+      return true;
+    }
+    for (Join join: joins) {
+      if (join.getQuery().leftFeatureGroupStartTime != null
```

Comment:
Why not call `.isTimeTravel` on the join.getQuery(). I think that would also make it real recursive, as right now only the first level of joins in checked.